bitcoin/src/network.rs

```diff
@@ -134,7 +197,8 @@ impl Network {
 
         let network = match core_arg {
             "main" => Bitcoin,
-            "test" => Testnet,
+            "test" => Testnet(TestnetVersion::V3),
+            "test4" => Testnet(TestnetVersion::V4),
```

Comment:
```suggestion
            "testnet4" => Testnet(TestnetVersion::V4),
```
bitcoind node will return `testnet4` and serialization will fail, you should use actual value instead of `test4`

![image](https://github.com/user-attachments/assets/0a91c75f-a6e9-40b2-97e3-d52c1fdf94bd)

```bash
curl --location --request POST 'http://127.0.0.1:48332' \
--header 'Authorization: Basic ...snip...' \
--header 'Content-Type: application/json' \
--data-raw '{
    "method": "getblockchaininfo",
    "params": []
}'
```

```
{
    "result": {
        "chain": "testnet4",
        "blocks": 36409,
        "headers": 36409,
        "bestblockhash": "0000000000000058e9f909a8d5a86afc8f58d39b661b02f07433d4cae8b6481d",
        "difficulty": 44472500.68822794,
        "time": 1721977342,
        "mediantime": 1721974212,
        "verificationprogress": 1,
        "initialblockdownload": false,
        "chainwork": "000000000000000000000000000000000000000000000048dcdd24a9fc06ef13",
        "size_on_disk": 284698607,
        "pruned": false,
        "warnings": [
            "This is a pre-release test build - use at your own risk - do not use for mining or merchant applications"
        ]
    },
    "error": null
}
```

Reply:
Insane, so the Core devs decided to make it as inconsistent as possible?

Reply:
PR is not yet finalized, but for now chain serialized as `testnet4` string
https://github.com/bitcoin/bitcoin/pull/29775/files/aba504759caa13338aa574a7839d078b989fbf6d#diff-254a3b28822b27e42585d1a3f4f41cc8d7712c911b7e85be78a2ee11564f4bd8R19